File: python/1716.计算力扣银行的钱.py
class Solution:
    # O(1) t:36ms(49%) m:14.9M(61%)
    def totalMoney(self, n: int) -> int:
        day = n % 7
        week = n // 7
        return 28 * week + week*(week+1)//2*7 + day*(day+1+2*week)//2

File: java/1716.计算力扣银行的钱.java
class Solution {
    // O(1) t:0ms(100%) m:35.4M(15%)
    public int totalMoney(int n) {
        int week = n / 7;
        int day = n % 7;
        return 28 * week + week*(week-1)*7/2 + day*(day+1+2*week)/2;
    }
}